subscriptions-api.md

```diff
@@ -0,0 +1,613 @@
+other settings SHOULD be supported.
+
+- **topicname** (string) – REQUIRED. The name of the MQTT topic to publish to.
+- **qos** (integer) – OPTIONAL. MQTT quality of service (QoS) level: 0 (at most
+  once), 1 (at least once), or 2 (exactly once). This defaults to 1 if not
+  set.
+- **retain** (boolean) – OPTIONAL. MQTT retain flag: true/false. This defaults to
+  false if not set.
+- **expiry** (integer) – OPTIONAL. MQTT expiry interval, in seconds. This value
+  has no default value and the message will not expire if the setting is
+  absent. This setting only applies to MQTT 5.0.
+- **userproperties** (map) – OPTIONAL. A set of key/value pairs that are copied into
+  the MQTT PUBLISH packet's user property section. This setting only applies
+  to MQTT 5.0.
+
+##### 3.2.2.3. AMQP
+
+For AMQP, the address property MUST be supported by all implementations and
+other settings properties SHOULD be supported by all implementations.
+
+- **address** (string) – OPTIONAL. The link target node in the
+  AMQP container identified by the sink URI, if not expressed in the sink URI's
+  path portion.
+- **linkname** (string) – OPTIONAL. Name to use for the AMQP link. If not set,
+  a random link name is used.
+- **sendersettlementmode** (string) – OPTIONAL. Allows to control the sender's
+  settlement mode, which determines whether transfers are performed "settled"
+  (without acknowledgement) or "unsettled" (with acknowledgement). Default
+  value is unsettled.
+- **linkproperties** (map) – OPTIONAL. A set of key/value pairs that are
+  copied into link properties for the send link.
+
+##### 3.2.2.4. Apache Kafka
+
+All implementations that support Apache Kafka MUST support the _topicname_
+settings. All other settings SHOULD be supported.
+
+- **topicname** (string) - REQUIRED. The name of the Kafka topic to publish to.
+- **partitionkeyextractor** (string) - OPTIONAL. A partition key extractor
+  expression per the CloudEvents Kafka transport binding specification.
+- **clientid** (string)
+- **acks** (string)
+
+##### 3.2.2.5. NATS
+
+- **subject** (string) - REQUIRED. The name of the NATS subject to publish to.
+
+#### 3.2.3 Filter Dialects
+
+The filter expression language supported by an event producer is indicated by
+its dialect. This is intended to allow for flexibility, extensibility and to
+allow for a variety of filter dialects without enumerating them all in this spec or
+predicting what filtering needs every system will have in the future. This spec
+will specify a single "simple" dialect, which all implementations MUST support.
+
+The dialect for a particular filter is indicated by specifying the "dialect"
+property at the root of the JSON object. The value of this is a string encoded
+unique identifier for the filter dialect. Subscriptions specifying the "filter"
+property MUST specify a dialect. All other properties are dependent on the
+dialect being used.
+
+##### 3.2.3.1. "simple" filter dialect
```

Comment:
simple does not describe what this dialect does, I would recommend renaming "simple" to "exact"

Reply:
"exact" isn't entirely accurate either though, as we allow for prefix and suffix matching. I don't really have a better suggestion than "simple", but I'm also not too particular here.

Reply:
I renamed it to "basic". That takes the simplicity connotation out.